src/mpc/mpc_dummy_phase2.cpp

```diff
@@ -78,20 +79,22 @@ class mpc_dummy_phase2 : public subcommand
         libff::leave_block("reading linear combination data");
 
         // Generate the zeth circuit (to determine the number of inputs)
+        libff::enter_block("computing num_inputs");
         const size_t num_inputs = [this]() {
             libsnark::protoboard<FieldT> pb;
             init_protoboard(pb);
-            const libsnark::r1cs_constraint_system<FieldT> cs =
-                pb.get_constraint_system();
-            return cs.num_inputs();
+            return pb.num_inputs();
         }();
+        libff::print_indent();
+        std::cout << std::to_string(num_inputs) << std::endl;
+        libff::leave_block("computing num_inputs");
 
-        // Generate the artifical delta
+        // Generate the artificial delta
```

Comment:
`Generate the artificial delta` -> `Generate a random delta locally to run the dummy phase2`?